test/job.js

```diff
@@ -765,21 +765,18 @@ describe('Job', () => {
     it('runs a recurring job after a lock has expired', async() => {
       let startCounter = 0;
 
-      // @TODO: this lint issue should be looked into: https://eslint.org/docs/rules/no-async-promise-executor
-      // eslint-disable-next-line no-async-promise-executor
-      const processorPromise = new Promise(async resolve =>
-        agenda.define('lock job', {
-          lockLifetime: 50
-        }, async() => {
+      const processorPromise = new Promise(resolve => {
+        agenda.define('lock job', { lockLifetime: 50 }, () => {
           startCounter++;
 
           if (startCounter !== 1) {
             expect(startCounter).to.be(2);
-            await agenda.stop();
-            resolve();
+            return agenda.stop().then(() => { // eslint-disable-line promise/prefer-await-to-then
```

Comment:
How come this needed `.then()`? 🤔 

Would be good to add a code comment to explain lint disable.

You might also shorten it to `.then(resolve)`

Reply:
changed to await style